src/agent/rustjail/src/mount.rs

```diff
@@ -578,12 +581,14 @@ fn parse_mount_table() -> Result<Vec<Info>> {
             String,
             String,
             String
-        )?;
+        )
+        .map_err(|_| anyhow!(ERR_FAILED_PARSE_MOUNTINFO))?;
 
         let fields: Vec<&str> = line.split(" - ").collect();
         if fields.len() == 2 {
             let (fstype, _source, _vfs_opts) =
-                scan_fmt!(fields[1], "{} {} {}", String, String, String)?;
+                scan_fmt!(fields[1], "{} {} {}", String, String, String)
+                    .map_err(|_| anyhow!(ERR_FAILED_PARSE_MOUNTINFO))?;
```

Comment:
It might be worth making this error message slightly "richer": the error shows the problem area, but it could state that the specific problem is that the optional mountinfo args are invalid, which would help debug any future issues in this area.

Reply:
Similar to my question below about optional fields. Would this indicate a problem with optional fields? Or are the last 3 items (9, 10, 11) as shown in this example optional, and I'm getting mixed up between those and optional fields (7)?

```
36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
(1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
```

Reply:
@bradenrayhorn - sorry, bad wording. I didn't mean optional fields, but the final fields. Currently, the code will fail here with the error "failed to parse mountinfo file", but we could be more specific and say something like, "failed to parse final fields in mountinfo file".